### PR TITLE
Improve documentation for `TaskAPI`

### PR DESCRIPTION
**Pull request by tortmayr**
Signed-off-by: Tobias Ortmayr <tortmayr@eclipsesource.com>
Contributed on behalf of STMicroelectronics

### CODE REVIEW

File: packages/task/src/node/task-runner.ts [hunk 1]
```diff
@@ -13,27 +13,44 @@
...
+     * Register custom runners using the given {@link TaskRunnerRegistry}.
+     * @param runners the common task runner registry.
+     */
     registerRunner(runners: TaskRunnerRegistry): void;
 }
 
 export const TaskRunner = Symbol('TaskRunner');
-/** A Task Runner knows how to run and kill a Task of a particular type. */
+/**
+ * A {@link TaskRunner} knows how to run and kill a task of a particular type.
```
Comment (koegel) [inline packages/task/src/node/task-runner.ts:35]: It does not know how to kill a task imho.

File: packages/task/src/node/task-runner.ts [hunk 2]
```diff
@@ -13,27 +13,44 @@
...
+     * @param taskConfig the task configuration that should be executed
+     * @param ctx the execution context
+     *
+     * @returns a promise of the (currently running) {@link Task}.
+     */
     run(taskConfig: TaskConfiguration, ctx?: string): Promise<Task>;
 }
 
+/**
+ * The {@link TaskRunnerRegistry} is the common componet for registration and provision of
```
Comment (koegel) [inline packages/task/src/node/task-runner.ts:49]:
```suggestion
 * The {@link TaskRunnerRegistry} is the common componet for the registration and provisioning of
```

File: packages/task/src/node/task-runner.ts [hunk 3]
```diff
@@ -49,21 +66,38 @@ export class TaskRunnerRegistry {
         this.runners = new Map();
         this.defaultRunner = this.processTaskRunner;
     }
-
-    /** Registers the given Task Runner to execute the Tasks of the specified type. */
+    /**
+     * Registers the given {@link TaskRunner} to execute Taks of the specified type.
```
Comment (koegel) [inline packages/task/src/node/task-runner.ts:70]:
```suggestion
     * Registers the given {@link TaskRunner} to execute Task of the specified type.
```

File: packages/task/src/node/task.ts [hunk 4]
```diff
@@ -18,13 +18,23 @@ import { injectable } from 'inversify';
...
-
+/**
+ * Represents the options used for running a task.
+ */
 export interface TaskOptions {
+    /** The task label */
     label: string;
+    /** The task configuration which should be executed */
     config: TaskConfiguration;
+    /** The context that can be used for aggregation */
```
Comment (koegel) [inline packages/task/src/node/task.ts:29]:
```suggestion
    /** The optional context that can be used for aggregation */
```

**Review by koegel — CHANGES_REQUESTED** [review]
Minor changes required

**Comment by tortmayr** [comment]
Thanks for the fast review. I addressed the issues and squashed the changes into a single commit

File: packages/task/src/node/task-runner.ts (hunk 1 above)
Comment (koegel) [reply packages/task/src/node/task-runner.ts:35]: Comment still open

**Review by koegel — CHANGES_REQUESTED** [review]
Open comment from last review

File: packages/task/src/node/task-runner.ts (hunk 1 above)
Comment (tortmayr) [reply packages/task/src/node/task-runner.ts]: Fixed

**Review by koegel — APPROVED** [review]

**Comment by sdirix** [comment]
Merged upstream with https://github.com/eclipse-theia/theia/pull/8695

**Pull request closed without merging**